### PR TITLE
Revert "[SPARK-25496][SQL] Deprecate from_utc_timestamp and to_utc_timestamp"

### PR DESCRIPTION
**Pull request by gatorsmile**
This reverts commit 1d20d13149140f53df307f47420740f45b4fa5f6.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DeprecatedDateFunctionsSuite.scala [hunk 1]
```diff
@@ -1,113 +0,0 @@
-/*
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/DeprecatedDateFunctionsSuite.scala]: The tests are moved by https://github.com/apache/spark/pull/26885

**Comment by dongjoon-hyun** [comment]
cc @MaxGekk and @cloud-fan

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Pending Jenkins. (This is a non-trivial clean revert which is related to multiple PRs. Most parts are the same the other adapted parts including StreamingaggregationSuite look good.)
Thank you, @gatorsmile .

**Comment by gatorsmile** [comment]
Thanks! @dongjoon-hyun 

Both Hive and Impala have such functions. We should simplify the migration by introducing the needed functions if we decide to drop/deprecate the existing functions. Adding a conf is not enough in such cases. 

I think we should introduce more time/timestamp related built-in functions in the future. Processing date and timestamp is a common task in the data processing/cleaning/analysis. We have to be more careful when introducing a behavior breaking change.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by HyukjinKwon — APPROVED** [review]
I am good with this way too since code freeze is close.
I think https://github.com/apache/spark/pull/24195#issuecomment-578046851 can be an option too.

**Comment by dongjoon-hyun** [comment]
Ping @MaxGekk and @cloud-fan once more since he is the original author and committer.

**Comment by MaxGekk** [comment]
@gatorsmile I do understand importance of backward compatibility but:
- Deprecated functions break semantic of the timestamp type. Its values are micros since the epoch which is in UTC timezone. Placing values of another meaning spreads errors across user apps unpredictably.
- You mentioned Hive/Impala. but if we looked at common used time API like Java 8 time API, Joda and etc. The libraries don't provide such functions.
- isn't the major version Spark 3.0 is good time to remove ugly functions from API? If you think we should put more details to Spark docs, let's do that.
- How else can we inform users that some functions have been deprecated already if we don't throw an exception, and force users to set SQL config, explicitly. At the moment, we don't have any mechanisms of informing users about deprecated functions, unfortunately. 

As summary, I cannot agree with reverting the changes. I would update docs and place a link to the docs into the exception message, and update expression description as well.

**Comment by dongjoon-hyun** [comment]
I'll merge this. We can deprecate later in two releases.
At the first release, Apache Spark should warn only as a log.
At the next release, Apache Spark can introduce a flag and switch the default.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Yeah, I think it's fine to revert it for now as it becomes pretty controversial. @MaxGekk I think we can open new PR as guided. Probably we should clearly document the alternatives.

**Pull request closed without merging**